src/settings/toml/manifest.rs

```diff
@@ -140,14 +174,7 @@ impl Manifest {
 
         target.route = self.negotiate_zoneless(environment)?;
         if let Some(environment) = environment {
-            target.name = if let Some(name) = &environment.name {
-                name.clone()
-            } else {
-                match environment_name {
-                    Some(environment_name) => format!("{}-{}", self.name, environment_name),
-                    None => failure::bail!("You must specify `name` in your wrangler.toml"),
-                }
-            };
+            target.name = self.worker_name(environment_name);
```

Comment:
here's that extracted method in use. note that this method does not have opinions about validity; that is evaluated elsewhere.